aws/resource_aws_iam_service_linked_role.go

```
@@ -0,0 +1,175 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsIamServiceLinkedRole() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIamServiceLinkedRoleCreate,
+		Read:   resourceAwsIamServiceLinkedRoleRead,
+		Delete: resourceAwsIamServiceLinkedRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"aws_service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, es []error) {
+					value := v.(string)
+					if !strings.HasSuffix(value, ".amazonaws.com") {
+						es = append(es, fmt.Errorf(
+							"%q must be a service URL e.g. elasticbeanstalk.amazonaws.com", k))
+					}
+					return
+				},
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"path": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"create_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsIamServiceLinkedRoleCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	serviceName := d.Get("aws_service_name").(string)
+
+	params := &iam.CreateServiceLinkedRoleInput{
+		AWSServiceName: aws.String(serviceName),
+	}
+
+	resp, err := conn.CreateServiceLinkedRole(params)
+
+	if err != nil {
+		return fmt.Errorf("Error creating service-linked role with name %s", serviceName)
```

We should return the error messages here and on deletion 😄  I will fix post-merge.